lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1718,6 +1746,22 @@ class MetricsEndpointAggregator(Object):
     """
 
     def __init__(self, charm, relation_names, relabel_instance=True):
+        """Construct a `MetricsEndpointAggregator`.
+
+        Args:
+            charm: a `CharmBase` object that manages this
+                `MetricsEndpointAggregator` object. Typically this is
+                `self` in the instantiating class.
+            relation_names: a dictionary with three keys. The value
+                of the "scrape_target" and "alert_rules" keys are
+                the relation names over which scrape job and alert rule
+                information is gathered by this `MetricsEndpointAggregator`.
+                And the value of the "prometheus" key is the name of
+                the relation with a `MetricsEndpointConsumer` such as
+                the Prometheus charm.
```

Comment:
I'd argue a [TypedDict](https://www.python.org/dev/peps/pep-0589/) would do an even better service :)

Reply:
Or a named tuple that way we can avoid the "Typed" part :-)